### PR TITLE
Documentation for ng-for

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: modules/angular2/src/core/directives/ng_for.ts [hunk 1]
```diff
@@ -13,17 +13,51 @@ import {isPresent, isBlank} from 'angular2/src/core/facade/lang';
...
+ * If the iterator itself is replaced, the DOM will be thrown out and recreated from scratch
+ * with elements from the new iterator.
+ *
+ * This has important implications for animations and any stateful controls in the template (such
+ * as `<input>` elements which accept user input). If the iterator contents change, Angular will
+ * reproduce those changes in the DOM. New rows can be animated in, removed rows can be animated
+ * out, and unchanged rows retain any unsaved state such as user input.
+ *
+ * If the entire iterator is replaced, Angular will tear down the entire DOM and rebuild it with
+ * from the new iterator, even if none of the elements have changed. This is an expensive
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_for.ts]: _with from_  -->  _with_ / _from_

File: modules/angular2/src/core/directives/ng_for.ts [hunk 2]
```diff
@@ -13,17 +13,51 @@ import {isPresent, isBlank} from 'angular2/src/core/facade/lang';
...
+ * operation, and should be avoided if possible.
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_for.ts]: OOC, are there any plans to add `track by` functionality to `ngFor` ?

File: modules/angular2/src/core/directives/ng_for.ts [hunk 3]
```diff
@@ -13,17 +13,51 @@ import {isPresent, isBlank} from 'angular2/src/core/facade/lang';
...
+ *
+ * If the iterator itself is replaced, the DOM will be thrown out and recreated from scratch
+ * with elements from the new iterator.
+ *
+ * This has important implications for animations and any stateful controls in the template (such
+ * as `<input>` elements which accept user input). If the iterator contents change, Angular will
+ * reproduce those changes in the DOM. New rows can be animated in, removed rows can be animated
+ * out, and unchanged rows retain any unsaved state such as user input.
+ *
+ * If the entire iterator is replaced, Angular will tear down the entire DOM and rebuild it with
```
Comment (vsavkin) [inline modules/angular2/src/core/directives/ng_for.ts]: Are you sure this is the case? If so, we actually need to change it, so it is not the case.

**Comment by IgorMinar** [comment]
@alxhub please clang-reformat and merge

File: modules/angular2/src/core/directives/ng_for.ts [hunk 4]
```diff
@@ -13,17 +13,52 @@ import {isPresent, isBlank} from 'angular2/src/core/facade/lang';
...
+ * * `index` will be set to the current loop iteration for each template context.
+ * * `last` will be set to a boolean value indicating whether the item is the last one in the
+ *   iteration.
+ * * `even` will be set to a boolean value indicating whether this item has an even index.
+ * * `odd` will be set to a boolean value indicating whether this item has an odd index.
+ *
+ * # Change Propagation
+ *
+ * There are two reasons why `NgFor` will need to update the DOM:
+ * * The contents of the iterator changes.
```
Comment (gkalpak) [inline modules/angular2/src/core/directives/ng_for.ts]: _contents_  -->  _content_ or _changes_  -->  _change_

**Pull request closed without merging**